### PR TITLE
{doc} Update domain after Microsoft Docs rebranding to Microsoft Learn

### PR DESCRIPTION
**Pull request by xmdanni**
**Related command**

**Description**

Microsoft Docs has been rebranded to Microsoft Learn for a long time, outdated links can have a negative impact on the output of the generative AI for Knowledge Service. So, links on reference pages need to use our new domain.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_proxy_resources_mgmt.yaml [hunk 1]
```diff
@@ -4002,7 +4002,7 @@ interactions:
...
+        of seconds to service waits before the binary log file gets purged. See: https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-server-parameters#binlog_expire_logs_seconds","defaultValue":"0","dataType":"Integer","allowedValues":"0-4294967295","source":"system-default","isConfigPendingRestart":"False","isDynamicConfig":"True"},"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.DBforMySQL/servers/azuredbclitest000002/configurations/binlog_expire_logs_seconds","name":"binlog_expire_logs_seconds","type":"Microsoft.DBforMySQL/servers/configurations"},{"properties":{"value":"0","description":"Controls
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/recordings/test_mysql_proxy_resources_mgmt.yaml:4005]: Recordings should be skipped. As previously motioned in this PR: https://github.com/Azure/azure-cli/pull/27040.

**Review by jiasli — APPROVED** [review]

**Review by AllyW — DISMISSED** [review]

**Review by AllyW — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**